test/integration_test.py

```diff
@@ -290,22 +292,42 @@ def __check_manifest(self, file: IO[bytes], uuid_field_name: str):
     def _download_with_drs(self, file_uuid: str):
         base_url = config.service_endpoint() + drs_http_object_path('')
         client = Client(base_url)
-        response = client.get_object(file_uuid)
-        self._validate_fastq_response_content(file_uuid, response.content)
-        log.info('Successfully downloaded file %s with DRS', file_uuid)
+        with self.subTest(access_method='https'):
+            response = client.get_object(file_uuid, access_method=AccessMethod.https)
+            self._validate_fastq_content(file_uuid, response.content)
+            log.info('Successfully downloaded file %s with DRS', file_uuid)
+        with self.subTest(access_method='gs'):
+            gs_url = client.get_object(file_uuid, access_method=AccessMethod.gs)
+            self.assertTrue(gs_url.startswith('gs://'))
+            if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
+                content = self.check_gs_url(gs_url)
+                self._validate_fastq_content(file_uuid, content)
+                log.info('Successfully downloaded from %s with DRS', gs_url)
+            else:
+                log.warning('Google credentials were not found via the '
+                            'GOOGLE_APPLICATION_CREDENTIALS environment '
+                            'variable, so the validity and content of the GS '
+                            'URL cannot be verified.')
+                self.assertFalse(config.is_main_deployment)
 
     def _download_with_dos(self, file_uuid: str):
         dos_endpoint = drs.dos_http_object_path(file_uuid)
         response = self._check_endpoint(config.service_endpoint(), dos_endpoint)
         json_data = json.loads(response)['data_object']
         file_url = first(json_data['urls'])['url']
-        response = self._check_endpoint(file_url, '')
-        self._validate_fastq_response_content(file_uuid, response)
+        content = self._check_endpoint(file_url, '')
+        self._validate_fastq_content(file_uuid, content)
         log.info('Successfully downloaded file %s with DOS', file_uuid)
 
-    def _validate_fastq_response_content(self, file_uuid, response):
+    def check_gs_url(self, url):
+        storage_client = storage.Client()
```

Comment:
See if you can force the client to only use SA creds from GOOGLE_APPLICATION_CREDENTIALS